Automattic/wp-calypso
# Decouple the Happy Chat button from the FAB and always show it when there’s an active chat session

This PR decouples the Happy Chat button from the FAB and makes sure that Happy Chat button is shown when a happychat session is active. Also, it changes the way that FAB was hidden in `home` section.

#### Changes proposed in this Pull Request

* Declouples Happy Chat button from the FAB
* "Hides" FAB by adding `home` in exempt sections of `shouldLoadInlineHelp()`
* Removed `/me/account/closed` as it redirects to `/me/account/close` and the FAB is already showing there

https://github.com/Automattic/wp-calypso/blob/11a93bf19bc86ae37bc16756d7c65bd706ed6130/client/layout/index.jsx#L144

#### Testing instructions

See p1614963839181400-slack-ajax for details on how to setup the HUD and chat to appear.



* Go to `https://hud-staging.happychat.io/` and login, making sure you can accept new chats
* Open a new tab in `http://calypso.localhost:3000/` 
* Make sure that after loading `/home` the FAB is not shown
* Go to `/stats` and confirm FAB is shown
* Start a new chat
* Go again in `/home` and confirm FAB is not shown but Happy Chat button is shown
* Reload the page, confirm that Happy Chat button is still showing
* Go to `/help` confirm that Happy Chat button is still showing
* Resize to a mobile breakpoint, clicking the Happy Chat icon should redirect to `/me/chat`

Before | After
-------|------
![](https://cln.sh/wLye3C+) | ![](https://cln.sh/H2xoW3+)



Fixes #50791
